skyportal/handlers/api/sources_confirmed_in_gcn.py

```diff
@@ -567,6 +570,146 @@ def delete(self, dateobs, source_id):
+            try:
+                sources_id_list = [
+                    source_id.strip() for source_id in sources_id_list.split(',')
+                ]
+            except ValueError:
+                return self.error(
+                    "some of the sourceIDs in the sourcesIDList are not valid strings"
+                )
+
+        sources_id_list = data.get('sourcesIDList', '')
+        if sources_id_list != '':
+            try:
+                sources_id_list = [
+                    source_id.strip() for source_id in sources_id_list.split(',')
+                ]
+            except ValueError:
+                return self.error(
+                    "some of the sourceIDs in the sourcesIDList are not valid strings"
+                )
+
+        with self.Session() as session:
+            try:
+                stmt = GcnEvent.select(session.user_or_token).where(
+                    GcnEvent.dateobs == dateobs
+                )
+                gcn_event = session.scalars(stmt).first()
+                if not gcn_event:
+                    return self.error(f"GCN event not found for dateobs: {dateobs}")
+
+                if len(sources_id_list) == 0:
+                    stmt = SourcesConfirmedInGCN.select(session.user_or_token).where(
+                        SourcesConfirmedInGCN.dateobs == dateobs
+                    )
+                else:
+                    stmt = SourcesConfirmedInGCN.select(session.user_or_token).where(
+                        SourcesConfirmedInGCN.dateobs == dateobs,
+                        SourcesConfirmedInGCN.obj_id.in_(sources_id_list),
+                    )
+                if confirmed:
+                    stmt = stmt.where(SourcesConfirmedInGCN.confirmed.is_(True))
+                sources_in_gcn = session.scalars(stmt).all()
+
+                tnsrobot = session.scalars(
+                    TNSRobot.select(session.user_or_token).where(
+                        TNSRobot.id == tnsrobotID
+                    )
+                ).first()
+                if tnsrobot is None:
+                    return self.error(f'No TNSRobot available with ID {tnsrobotID}')
+
+                altdata = tnsrobot.altdata
+                if not altdata:
+                    return self.error('Missing TNS information.')
+                if 'api_key' not in altdata:
+                    return self.error('Missing TNS API key.')
+
+                try:
+                    loop = asyncio.get_event_loop()
+                except Exception:
+                    loop = asyncio.new_event_loop()
+                    asyncio.set_event_loop(loop)
+
+                IOLoop.current().run_in_executor(
+                    None,
+                    lambda: tns_submission(
+                        [obj.obj_id for obj in sources_in_gcn],
+                        tnsrobot.id,
+                        self.associated_user_object.id,
+                        reporters=reporters,
+                    ),
+                )
```

Comment:
Yes, I'll use it to query the error now. The id is `133291`

Reply:
Ok so its because the reporters was empty, like I didn't specify it in the request. What should be the behavior here you think? We create one using user name and last name and group, or we make it mandatory to specify? I think the second makes more sense, because users don't all have name and lastname setup, and the group might be like `EM+GW` when really they want to report for ZTF. Let me know what you think and I'll add that code

Reply:
@Theodlz I think we should make it required.

Reply:
@mcoughlin making sure reporters isnt empty (by making it mandatory to pass) fixed it 👍